### ISSUE TITLE
Recently can't stay logged in - Your blue iris session has expired. Attempting recovery

### ISSUE THREAD

**Issue by ahaverty** [bp2008/ui3#162]
For the last few days, after logging in, I can see my cameras, but get repeated warning dialogues bottom right:
"Your blue iris session has expired. Attempting recovery"

They seem to launch dozens of times until it eventually  forces me back out to the login screen automatically, sometimes saying something too quick to read again about session expiry.

If I try to login a few more times, I can see my cameras live views, but it'll eventually give me a red warning on login: "connection limit, try again later"

I've tried both with login automatically enabled and disabled.

I'm not sure where to even start debugging. I updated to the latest Blue Iris yesterday to see if that would fix it (it hasn't). And I've tried totally deleting cache/local storage for my ui3 site.

**Comment by bp2008** [comment]
That is weird.  I suspect something is interfering with the session cookie.

Look at the Blue Iris Status window, Connections tab to see active and recent connections.  I expect you will find it quickly filling up when you try this.

In Blue Iris Settings > Web server > Advanced, there is a setting for a Connection limit.  Default is 99, make sure it didn't get lowered somehow.

If you host any other web sites at the same URL, they might also be using an HTTP cookie named "session" which interferes with the one used by Blue Iris.

Otherwise the next suspects would be:

* browser extensions (try using an incognito window as that should not have any extensions in it)
* browser privacy settings
* Any proxy server you may be using between the client device and Blue Iris
* Security software on the client device interfering with outbound HTTP traffic
* Security software on the BI server interfering with inbound HTTP traffic

It is also a very long shot, but you could try rebooting the client device in case it has a glitched out browser instance in the background doing something unexpected.

**Comment by bp2008** [comment]
> (I wonder has the latest release of ui3 got an issue where it's refreshing a lot sooner than 3 seconds like the notes state?)

Nah, the 3 second refresh would be a full page reload and it would only happen if one of those loading status indicators changed to "FAIL".  What is happening to you there is UI3 is detecting that the session is bad, so it logs you back in and retries whatever had failed. But that fails again for the same reason, so it loops until your session count is maxed out.

> Any idea if there's a simple workaround for the session naming?

The best solution would be to use a different subdomain for Blue Iris access, one that no other web applications use.


Ideally, Blue Iris would have used the name "bisession" or something for the cookie.  It would be a little problematic if they were to change the cookie name now because some third-party integrations (like UI3 itself) expects it to be named "session".


Alternatively you could try disabling cookies for the domain in question.  UI3 should be able to operate with cookies disabled, but with reduced loading performance because caching won't work.  But the other applications you host on the same domain would also be forced to work without cookies and they probably haven't been coded to support that.

**Comment by bp2008** [comment]
Yeah ok so disabling cookies also disabled localStorage (which is where UI3 stores all of its settings).  That makes it a no-go.

Very weird that enabling SSL would fix it.

I've looked into the related UI3 source code.  I no longer suspect it has anything to do with cookies.  Based on the video you showed, here is what I think was happening:

During UI3 startup, UI3 tries to load the camera list using Blue Iris's JSON API.  In response, Blue Iris sends a JSON-encoded object with a property named "result" which has the value "success" or "fail". In this case, it is "fail", which UI3 has long understood to mean the session was lost and it needs to log in again.

Except in this particular situation, I'm thinking the "fail" result must have a different cause than UI3 expects.  Because when UI3 handles it by creating a new session and trying to request the camera list again, it gets back the same "fail" result again.  It continues like this in a loop until you have 99 active sessions, then it can't log in another one and it fails due to the connection limit.

The problem here is that UI3 treats every camera list "fail" result as if it was caused by a session loss, because when it IS truly caused by a bad session, Blue Iris doesn't say so. 
 It just says it failed and gives no reason.  So UI3 pretty much has no choice but to assume it is caused by session loss because that is the real cause in most cases.

I'll open a support ticket with Blue Iris support to deal with this.

**Comment by rrmelgar** [comment]
getting this issue too 
![Screenshot_20241003_140133_Brave](https://github.com/user-attachments/assets/c3b22907-c1c4-49ca-8d52-fcf7724ccdfe)
![Screenshot_20241003_140135_Brave](https://github.com/user-attachments/assets/25ec9b26-8da6-4f21-ac6b-2d338a507fad)
all browser affected this issue android,apple
